services/rfq/relayer/service/handlers.go

```diff
@@ -81,11 +93,23 @@ func (r *Relayer) handleBridgeRequestedLog(parentCtx context.Context, req *fastb
 		Transaction:         bridgeTx,
 		Status:              reldb.Seen,
 		OriginTxHash:        req.Raw.TxHash,
-	})
+	}
+	err = r.db.StoreQuoteRequest(ctx, dbReq)
 	if err != nil {
 		return fmt.Errorf("could not get db: %w", err)
 	}
 
+	// immediately forward the request to handleSeen
+	span.AddEvent("forwarding to handleSeen")
+	qr, err := r.requestToHandler(ctx, dbReq)
+	if err != nil {
+		return fmt.Errorf("could not get quote request handler: %w", err)
+	}
+	err = qr.handleSeen(ctx, span, dbReq)
```

Comment:
This method has two issues:

1. calling the method like this avoids the deadline middleware and the gas middleware
2. calling this method create's a race condition that can result in two relays being submitted at the same time.

WRT 1., just call qr.Handle() instead of handle Seen

WRT 2. As @aureliusbtc pointed out in slack, this creates a race condition where we can submit two "Relay"'s at the same time

I'd recommend creating a new middleware we apply to all functions that uses [stringmutex](https://pkg.go.dev/github.com/synapsecns/sanguine/core/mapmutex#StringMapMutex) to lock each tx so it won't get processed more than once

Reply:
Hmm, if we implement the mutex as a middleware don't we lose all the concurrency gains? Since each request handling will have to be in serial?

What if we just put a lock around submitter to make sure SubmitTransaction() can only be called once concurrently, while the rest of the request handling routine can be done in parallel?

Agree that we can also separate the processDB() loops into pre and post-relay txs